lib/net.c

```diff
@@ -824,82 +953,22 @@ gatekeeper_get_rss_config(uint16_t port_id,
 
 static int
 init_port(struct gatekeeper_if *iface, uint16_t port_id,
-	uint8_t *pnum_succ_ports)
+	const struct rte_eth_conf *port_conf)
 {
-	struct rte_eth_conf port_conf = gatekeeper_template_port_conf;
-	struct rte_eth_dev_info dev_info;
-	int ret;
-
-	rte_eth_dev_info_get(port_id, &dev_info);
-
-	if (dev_info.flow_type_rss_offloads != 0) {
-		uint64_t configured_rss_hf;
-
-		iface->rss = true;
-
-		port_conf.rxmode.mq_mode = ETH_MQ_RX_RSS;
-
-		port_conf.rx_adv_conf.rss_conf.rss_key = default_rss_key;
-		port_conf.rx_adv_conf.rss_conf.rss_key_len =
-			GATEKEEPER_RSS_KEY_LEN;
-		port_conf.rx_adv_conf.rss_conf.rss_hf = ETH_RSS_IP;
-
-		/* Only use RSS hash functions this device can handle. */
-		configured_rss_hf = port_conf.rx_adv_conf.rss_conf.rss_hf;
-		port_conf.rx_adv_conf.rss_conf.rss_hf &=
-			dev_info.flow_type_rss_offloads;
-
-		/*
-		 * Check whether this device supports the configured RSS hashes.
-		 *
-		 * It seems common that devices do not exactly support the
-		 * hashes in the DPDK macros such as ETH_RSS_IP, so until we
-		 * choose set of minimum hash functions required (instead of
-		 * ETH_RSS_IP which is overkill), issue a warning in that case.
-		 *
-		 * TODO #152 Find the minimum set of hash functions (ETH_RSS_*) that
-		 * Gatekeeper needs and set
-		 * gatekeeper_port_conf.rx_adv_conf.rss_conf.rss_hf accordingly.
-		 * Then, change this warning to an error.
-		 */
-		if (configured_rss_hf !=
-				port_conf.rx_adv_conf.rss_conf.rss_hf) {
-			G_LOG(WARNING,
-				"net: port %hu invalid configured rss_hf: 0x%"PRIx64", valid value: 0x%"PRIx64"\n",
-				port_id, configured_rss_hf,
-				port_conf.rx_adv_conf.rss_conf.rss_hf);
-		}
-	} else {
-		G_LOG(WARNING, "net: the %s interface does not have RSS capabilities; the GK or GT block will receive all packets and send them to the other blocks as needed. Gatekeeper or Grantor should only be run with one lcore dedicated to GK or GT in this mode; restart with only one GK or GT lcore if necessary\n",
-			iface->name);
-		iface->num_rx_queues = 1;
-	}
-
-	port_conf.rxmode.max_rx_pkt_len = iface->mtu;
-
-	/*
-	 * If the MTU is set above the mbuf segment size, then hardware
-	 * support for transmitting multiple segments should be enabled.
-	 */
-	if (iface->mtu > RTE_MBUF_DEFAULT_BUF_SIZE)
-		port_conf.txmode.offloads |= DEV_TX_OFFLOAD_MULTI_SEGS;
-
-	ret = rte_eth_dev_configure(port_id, iface->num_rx_queues,
-		iface->num_tx_queues, &port_conf);
+	int ret = rte_eth_dev_configure(port_id, iface->num_rx_queues,
+		iface->num_tx_queues, port_conf);
 	if (ret < 0) {
-		G_LOG(ERR, "net: failed to configure port %hhu (err=%d)\n",
+		G_LOG(ERR, "net: failed to configure port %hu (err=%d)\n",
 			port_id, ret);
 		return ret;
 	}
-	if (pnum_succ_ports != NULL)
-		(*pnum_succ_ports)++;
-
 	return 0;
 }
 
 static int
 init_iface(struct gatekeeper_if *iface)
 {
+	struct rte_eth_conf port_conf = gatekeeper_template_port_conf;
 	int ret;
 	uint8_t i;
 	uint8_t num_succ_ports = 0;
```

Comment:
1. Style: move initialization of variable `num_succ_ports` to where it's needed. This helps the compiler to help us.

2. Style: Move the declaration of variable `num_slaves_added` (I can't comment at its line) to the code block where it's used. And its initialization where it's needed.